packages/opentelemetry-context-async-hooks/test/AsyncHooksContextManager.test.ts

```diff
@@ -15,339 +15,365 @@
  */
 
 import * as assert from 'assert';
-import { AsyncHooksContextManager } from '../src';
+import {
+  AsyncHooksContextManager,
+  AsyncLocalStorageContextManager,
+} from '../src';
 import { EventEmitter } from 'events';
 import { Context } from '@opentelemetry/context-base';
 
-describe('AsyncHooksContextManager', () => {
-  let contextManager: AsyncHooksContextManager;
-  const key1 = Context.createKey('test key 1');
-
-  beforeEach(() => {
-    contextManager = new AsyncHooksContextManager();
-    contextManager.enable();
-  });
-
-  afterEach(() => {
-    contextManager.disable();
-  });
-
-  describe('.enable()', () => {
-    it('should work', () => {
-      assert.doesNotThrow(() => {
-        contextManager = new AsyncHooksContextManager();
-        assert(
-          contextManager.enable() === contextManager,
-          'should return this'
-        );
-      });
-    });
-  });
-
-  describe('.disable()', () => {
-    it('should work', () => {
-      assert.doesNotThrow(() => {
-        assert(
-          contextManager.disable() === contextManager,
-          'should return this'
-        );
-      });
+for (const contextManagerClass of [
+  AsyncHooksContextManager,
+  AsyncLocalStorageContextManager,
+]) {
+  describe(contextManagerClass.name, () => {
+    let contextManager:
+      | AsyncHooksContextManager
+      | AsyncLocalStorageContextManager;
+    const key1 = Context.createKey('test key 1');
+
+    beforeEach(function () {
+      if (
+        contextManagerClass.name === 'AsyncLocalStorageContextManager' ||
```

Comment:
i think this should be `contextManagerClass.name === 'AsyncLocalStorageContextManager' && (process.version.startsWith('8.') || process.version.startsWith('10.'))`